go/src/url_shortener/url_shortener.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"crypto/sha1"
+	"fmt"
+	"log"
+	"net/http"
+	"sync"
+)
+
+var (
+	cache URLShortener = URLShortener{shorts: make(map[string]*urlInfo)}
```

Comment:
At the moment it's not a problem to use a global for this. However it might be when you're going to add some unit-tests (in a successive issue).